### DISCUSSION TITLE
Removing or Cropping out the border from scanned images.

### DISCUSSION THREAD

**Discussion by sbudhdeo** [ImageMagick/ImageMagick#6552] [Digital Image Processing]
Hello, please see the attached image for reference.  I have thousands of scanned postcards done with a duplex scanner.  The scanner itself cannot fully remove the grey border as shown.  Is there a way with IM that I'd be able to remove the grey border or replace it with white, or crop it out? If possible, I'd be doing this in batches.
![sample](https://github.com/ImageMagick/ImageMagick/assets/7156122/ebd34493-c910-411e-bd1c-e7286696e7c2)

**Comment by fmw42** [comment]
Please always answer the questions about your IM version and platform.  Assuming IM 6, you can trim (remove) the gray using:

`convert image.jpg -fuzz 45% -trim +repage trimmed.jpg`

or for IM 7

`magick image.jpg -fuzz 45% -trim +repage trimmed.jpg`

![trimmed](https://github.com/ImageMagick/ImageMagick/assets/13458218/dab3008a-1c37-4ee7-8b85-65982c365cd8)


Adjust the fuzz value for how uneven the gray color is.  Very uniform might be -fuzz 0. But typically even for uniform color, JPG compression artifacts require a minimum of about 15%.  Your gray is not very uniform so I had to increase to about 45%.  Too high and it will remove other colors. Too low and it will not remove enough gray (outer border color).

If you need the black at the bottom removed, the easiest would be to simply -chop it off. However, you can add a black border all around and then -trim again.

If you need to process a number of images with the same gray border, you can try using mogrify (IM 6) or magick mogrify for IM 7.  See https://imagemagick.org/Usage/basics/#mogrify.  Alternately, write a script FOR loop over each image and use the convert / magick command above.
